src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useState } from "react";
+import "./Modal.css";
+import ImageUploading, { ImageListType } from "react-images-uploading";
+import { Post } from "../App/App";
+// import { idText } from "typescript";
+
+interface ModalProps {
+  submitPost: (newPost: Post) => void;
+  closeModal: (event: any) => void;
+}
+
+export const Modal: React.FC<ModalProps> = ({ submitPost, closeModal }) => {
+  const [images, setImages] = useState([]);
+  const [postTitle, setTitle] = useState<string>("");
+  const [postDescription, setDescription] = useState<string>("");
+  const [imageURLS, setImageURLS] = useState<string[]>([]);
+
+  const maxNumber = 2;
+  const onChange = (
+    imageList: ImageListType,
+    addUpdateIndex: number[] | undefined
+  ) => {
+    setImages(imageList as never[]);
+    let onlyURLs = imageList.map((image) => image.data_url);
+    setImageURLS(onlyURLs);
+  };
+
+  const clearState = () => {
+    setImages([]);
+    setTitle("");
+    setDescription("");
+  };
+
+  const addPost = (event: React.MouseEvent<HTMLElement>) => {
+    event.preventDefault();
+    const newPost = {
+      id: 2,
+      title: postTitle,
+      description: postDescription,
+      image: imageURLS,
+      user: { id: 4, username: "workerBee1" },
+      upvotes: 0,
+      downvotes: 0,
+      created_at: "August 29, 2021",
```

Comment:
Curious here, too -- we'll add a created_at date, so unless we're worried that we need the created_at date to be the exact millisecond the post is created on the FE, you can send us the post info without needing to send created_at at all.

Reply:
Yeah, I'd remove it actually. It's best for the database to time stamp the creation of the record

Reply:
Yeah this will come out once we switch to graphQL🤘

Reply:
For the sake of keeping TS happy, We are going to keep it for now and remove later.